### ISSUE TITLE
CephFS volumes need --id argument to mount correctly with ceph-fuse 

### ISSUE THREAD

**Issue by larryck** [kubernetes/kubernetes#59393]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
I tried to use pvs provisioned by the cephfs storageclass build from https://github.com/kubernetes-incubator/external-storage/tree/master/ceph/cephfs. PV and pvc are OK but it failed when mounting. here is the output of the ceph-fuse mount command:

> ceph-fuse[607026]: starting ceph client2018-02-06 11:51:02.441782 7fb3adf000 -1 init, newargv = 0x1745f9710 newargc=11
> 
> ceph-fuse[607026]: ceph mount failed with (22) Invalid argument
> ceph-fuse[607024]: mount failed: (22) Invalid argument

I try the ceph-fuse mounting outside kubernetes environment, and found that I have to pass the --id argument to make ceph-fuse work correctly:

> root@node1:~/ck_pls_keep# ceph-fuse -k /mnt/ceph.client.tst1.keyring /mnt/test-fuse/ -r /tst1
> ceph-fuse[608668]: starting ceph client2018-02-06 11:55:49.190379 7fa0772000 -1 init, newargv = 0x16cdcb710 newargc=11
> 
> ceph-fuse[608668]: ceph mount failed with (22) Invalid argument
> ceph-fuse[608666]: mount failed: (22) Invalid argument

> root@node1:~/ck_pls_keep# ceph-fuse --id tst1 -k /mnt/ceph.client.tst1.keyring /mnt/test-fuse/ -r /tst1
> ceph-fuse[608691]: starting ceph client
> 2018-02-06 11:55:55.997967 7f7bb25000 -1 init, newargv = 0x148c19710 newargc=11
> ceph-fuse[608691]: starting fuse

so I add the --id option in file pkg/volume/cephfs/cephfs.go on line 401:
```
        src += hosts[i]

        mountArgs := []string{}

        **//add by tanck
        mountArgs = append(mountArgs, "--id")
        mountArgs = append(mountArgs, cephfsVolume.id)**

        mountArgs = append(mountArgs, "-k")

        mountArgs = append(mountArgs, keyring_file)
```
and cephfs volumes can mount corretly with ceph-fuse then.

**Anything else we need to know?**:
My kubernetes cluster version is v1.6.7, running on arm64 servers, with ceph version 10.1.2, the kernel version is 4.4.58 . 
I backported the latest pkg/volume/cephfs/cephfs.go version to mount cephfs volumes with ceph-fuse.

**Comment by mlmhl** [comment]
Does this problem still exist in the latest k8s version?

**Comment by larryck** [comment]
yes, I've tested on kubernetes v1.10.0-alpha.3 with  ceph version 10.2.10 and the problem still exists

**Comment by rootfs** [comment]
@zhangxiaoyu-zidif

**Comment by zhangxiaoyu-zidif** [comment]
OK, let me see it ,and feedback soon. =)

**Comment by zhangxiaoyu-zidif** [comment]
@larryck  Hi, one question is that do you use 'admin' as user or other user to mount cephfs by fuse ?

**Comment by zhangxiaoyu-zidif** [comment]
https://github.com/kubernetes/kubernetes/blob/c8841cea3092f0ef05d85b8686e552c93a040ec8/pkg/volume/cephfs/cephfs.go#L125

when no "--id" specified, the default user is admin, which will result in bug when user is not admin.
